Update Haskell package set to Stackage Nightly 2021-03-17 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-03-19 20:00 UTC+1**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other.

peti committed:
haskellPackages.{qtah*,hoppy*}: Fix build and mark as unbroken

Anton-Latukha committed:
haskellPackages.hnix-store-core: 0.4.2.0

Release just was made, fixed the `tasty-discover`.

peti committed:
ghc: drop obsolete version 8.6.5

peti committed:
ghc: drop unused binary compiler version 8.2.2

peti committed:
ghc: drop obsolete configuration for version 8.4.x

peti committed:
ghcjs: drop obsolete configuration file

peti committed:
ghc: drop obsolete compiler version 8.8.2

peti committed:
ghc: drop obsolete compiler version 8.8.3

peti committed:
ghc: drop obsolete compiler version 8.10.1

peti committed:
ghc: drop obsolete compiler version 8.10.2

peti committed:
ghc: drop obsolete compiler version 8.10.3

peti committed:
all-cabal-hashes: update to Hackage at 2021-03-12T19:55:03Z

peti committed:
haskell-language-server: drop obsolete ghc-8.6.5 from the wrapper

peti committed:
shake-bench: remove 'broken' mark

This package has a maintainer.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/bab9a536691c2280293100b200064dd3c3b01ddf.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/e6dd41d4c281346deaf7e9122df44bc9542b7700.

cdepillabout committed:
Merge pull request #114791 from turboMaCk/haskell-fix-qtah

haskellPackages.qtah*: Fix build and mark as unbroken

cdepillabout committed:
Merge pull request #116089 from Anton-Latukha/update

haskellPackages.hnix-store-core: 0.4.2.0

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/2fadba598a4b0b91e6b5156894e9a79919dd4d9f.

bennofs committed:
haskell-network_2_6_3_1: disable test suite

The package is missing required files for the test suite. This fixes the
following build error:

```
  Building test suite 'regression' for network-2.6.3.1..
  [1 of 1] Compiling Main             ( tests/Regression.hs,
dist/build/regression/regression-tmp/Main.o )

  tests/Regression.hs:12:1: error:
      Could not find module ‘Regression.Issue215’
      Use -v (or `:set -v` in ghci) to see a list of the files searched
for.
     |
  12 | import qualified Regression.Issue215 as Issue215
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

cdepillabout committed:
Merge pull request #116428 from bennofs/haskell-old-network-notest

haskell-network_2_6_3_1: disable test suite

peti committed:
Stackage Nightly 2021-03-14

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/86da7f5ea38187d604238f3c183b0c37a3862864.

sternenseemann committed:
haskellPackages.spacecookie: enable extra tests, install man pages

Add an override to configuration-common.nix adding the following
features to the derivation:

* Let test suite discover the built spacecookie binary, so it doesn't
  skip integration tests (which are very cheap and take just over 1s).
* Install man pages shipped in the sdist. (If someone is eager to get
  rid of this override feel free to explain to me how to achieve this
  without a Custom build-type which pulls in thousands of modules from
  Cabal. :p)

cdepillabout committed:
Merge pull request #116560 from sternenseemann/spacecookie-1.0.0.0-overrides

haskellPackages.spacecookie: enable extra tests, install man pages

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/5d8fd28c580916a0b9205ee3e1964eeb438147de.

peti committed:
Stackage Nightly 2021-03-17

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/5d8fd28c580916a0b9205ee3e1964eeb438147de.

peti committed:
Merge origin/master into haskell-updates.

Thra11 commented:
I assume with the newly added aarch64 support, we now need to track down all the packages which _shouldn't_ be supported on aarch64 (such as the macOS-specific 'barbly' which tripped ofborg up), and add them to `unsupported-platforms` in `configuration-hackage2nix.yaml`.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/6a804ff314aab4827da4ec3b03c0218d7b4099ed.

peti committed:
optparse-applicative: drop obsolete patches

peti committed:
cabal-install-parsers: drop obsolete overrides to fix evaluation

peti committed:
update-nix-fetchgit: drop obsolete overrides to fix evaluation

peti committed:
cryptohash-{sha1,md5}: disable test suites to fix the build

peti committed:
hackage2nix: update list of broken builds to fix evaluation errors on Hydra

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/1385c34be7c95d18840b3094d89cb9030a958796.

peti committed:
haskell-zlib: re-enable the test suite -- it builds fine now

peti committed:
text-short: jailbreak to allow building with recent versions of tasty

peti committed:
resolv: jailbreak to fix the build with recent versions of base16-bytestring

peti committed:
hackage2nix: keep base16-bytestring-0.x around in the package set

We need it to build cabal-install.

peti committed:
cabal-install: fix overrides for ghc-8.10.x

peti committed:
configuration-ghc-8.10.x.nix: cosmetic

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/1385c34be7c95d18840b3094d89cb9030a958796.

peti committed:
tdigest: jailbreak to allow building with recent versions of tasty

peti committed:
Drop erroneous dontDistribute overrides. These packages should be marked broken.

peti committed:
tree-diff: jailbreak to allow building with recent versions of tasty

peti committed:
cbord: jailbreak to allow building with recent versions of tasty

We also need the 0.x version of base16-bytestring to build successfully.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/1385c34be7c95d18840b3094d89cb9030a958796.

sternenseemann committed:
haskellPackages.chatter: unbreak

chatter's latest hackage release still depends on regex-tdfa-text, but
we can apply a patch from master to remove that dependency and jailbreak
to relax the bounds on cereal. Both these issues are already resolved
on master, so the override should only stand until the next release
comes around.

Additionally the test suite needs disabling as it doesn't list all
required modules in other-modules and thus fails to compile. The issue
has been reported upstream.

haskellPackages.fullstop: unbreak

Unfortunately fullstop is practically unmaintained and has no issue
tracker. The build failure is fortunately only affecting the test suite,
so a dontCheck resolves the issue for now.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/66b650498e63c1635ae6210184069c39654769f5.

cdepillabout committed:
Merge pull request #116803 from sternenseemann/chatter-unbreak

haskellPackages.chatter: unbreak

peti committed:
haskeline: update override for the new version

maralorn committed:
haskell-language-server: Fix build

maralorn committed:
Merge pull request #116841 from maralorn/hls

haskell-language-server: Fix build

peti committed:
haskell-aws: add jailbreak to fix the build with base16-bytestring version 1.x

peti committed:
hackage2nix: disable Windows-only builds an aarch64-linux and armv7l-linux

peti committed:
pandoc: disable the test suite to fix the build

It seems like a recent update of a dependency has broken ~7 tests in Pandoc.
Upstream is aware of the issue.

peti committed:
cabal-install: fix the build with ghc-8.8.x

peti committed:
cabal-install: fix the build with ghc-9.0.x

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/15683ff62a08f76d2c7c049ac6d99f77b6f892e6.

peti committed:
all-cabal-hashes: update to Hackage at 2021-03-19T20:01:05Z

peti commented:
We'll merge https://github.com/NixOS/nixpkgs/pull/116965 next week.
